src/server/project.ts

```diff
@@ -288,6 +293,7 @@ namespace ts.server {
                 hasChanges = this.updateGraphWorker() || hasChanges;
             }
             if (hasChanges) {
+                this._isJsOnlyProject = allRootFilesAreJsOrDts(this);

```

Comment:
actually after thinking about it a bit more better places to invalidate the value will be `addRoot` and `removeRoot` - there we can set `this._isJsOnlyProject` to `undefined` and compute the value on demand when it is requested